### DISCUSSION TITLE
Search metrics

### DISCUSSION THREAD

**Discussion by jmooring** [jameslittle230/stork#259] [Q&A]
We implement site search to help our visitors rapidly find what they are looking for, but we also need to know what they're looking for in order to improve content/indexing over time (among other things).

Has anyone come up with something to accomplish this?

For example, a hook of some sort to send an event to Google Analytics via gtag.js? From my very limited understanding of how Stork operates under the hood, it seems like this would have to be handled by the WebAssembly file and some configuration options that are stored in the index.

**Comment by jameslittle230** [comment]
The Stork Javascript API has hooks that might allow you to perform the tracking you want to! Depending on what you want to track, you could send an event via the [`onQueryUpdate`](https://stork-search.net/docs/js-ref#onQueryUpdate) callback function (to log whenever someone types something into the search box), or you could send an event via the [`onResultSelected`](https://stork-search.net/docs/js-ref#onResultSelected) callback to log whenever someone selects a search result.

I use both in the Stork documentation site (you can see the source [here](https://github.com/stork-search/site/blob/master/src/components/stork.js#L36-L47).) I use a self-hosted [Matomo](https://matomo.org) installation instead of Google Analytics, but I assume the Google Analytics API is similar.

Let me know if that matches up with what you were looking for!

-James
